src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -111,13 +114,49 @@ def get_disk(self, name):
         except pyudev.DeviceNotFoundByNameError:
             return None
 
-        return self.get_disk_details(block_device, disk, self.retrieve_lshw_disks_data())
+        return self.get_disk_details(block_device, disk, self.retrieve_lsblk_disks_data())
 
     @private
-    def get_disk_details(self, block_device, disk, lshw_disks):
+    def get_rotational_rate(self, device_path):
+
+        rotation_rate = None
+
+        vpd = subprocess.Popen(
+            ['sg_vpd', '-p', 'bdc', device_path],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        out, err = vpd.communicate()
+        if not vpd.returncode:
+            out = out.decode().split('\n')[1].strip()
```

Comment:
How about using regex or checking key name like `Nominal rotation rate` rather then relying on `1` index as it could probably change ? We are already checking `rotation rate` in if below but we might lose it here before as it might on a different index

Reply:
sg_vpd has a very stable API. If this was the case, then we would also need to do a regex for the lunid section. I'd rather leave this as is.

Reply:
Why the if check below then ? ; )
My concern was that we do a check partially, so it shows we are not sure of the output, but doing the check there means we already have potentially lost the correct line.

Reply:
Because `Nominal rotation rate:` doesn't get returned on SSD's.

Reply:
Even though we're only doing this on HDD types, having that extra `if` is to make sure that `lsblk` identified the type correctly and passed us a HDD.

Reply:
I don't like it, but to get your approval I've changed it to use a REGEX 👍

Reply:
Thank you for clarifying and using regex @yocalebo : ) I was okay with it if you think it's good :p